File: FromSlides/Fortran/22_set_threads_in_directive.f90
PROGRAM hello_world

  USE omp_lib
  INTEGER :: n_threads = 6

  PRINT *, 'Hello world from the main program'

  !$OMP PARALLEL NUM_THREADS(n_threads)
    PRINT '(A,I0)','Hello world! I am processor ', omp_get_thread_num()
  !$OMP END PARALLEL

END PROGRAM hello_world

File: FromSlides/C/22_set_threads_in_directive.c
#include <stdio.h>
#include <omp.h>

int main()
{
  printf("Hello world from the main program\n");

  int nthreads = 6;
#pragma omp parallel num_threads(nthreads)
  {
    printf("Hello world! I am processor %i\n", omp_get_thread_num());
  }
  return 0;
}

